src/malli/core.cljc

```diff
@@ -547,6 +547,27 @@
                          x))))
      :cljs (fn [x] (into (when x empty) (map t) x))))
 
+(defn -or-transformer [this transformer child-schemas method options]
+  (let [this-transformer (-value-transformer transformer this method options)]
+    (if (seq child-schemas)
+      (let [transformers (-vmap #(or (-transformer % transformer method options) identity) child-schemas)
+            validators (-vmap -validator child-schemas)]
+        (-intercepting this-transformer
+                       (if (= :decode method)
+                         (fn [x]
+                           (key (reduce-kv
+                                 (fn [acc i transformer]
+                                   (let [x* (transformer (val acc))]
+                                     (if ((nth validators i) x*)
+                                       (reduced (miu/-tagged x* nil))
+                                       (miu/-tagged (or (key acc) x*) x))))
+                                 (miu/-tagged nil x) transformers)))
```

Comment:
This loop feels too complicated for what it's doing. It looks like `(val acc)` is always just the original `x`. It also looks like this could just be an application of `some` instead of a `reduce-kv`.

Reply:
good catch, the `miu/-tagged` was not needed! I don't think `some` is best here as:
1) we have to memoize the first value (the fallback value)
2) some uses sequence abstraction, which is much slower

Reply:
committed a [simpler loop](https://github.com/metosin/malli/pull/946/commits/85de30f2c118dad854e4244fae88413e65fe7392).